src/preferences/dialog/dlgprefeffects.cpp

```diff
@@ -315,28 +315,40 @@ bool DlgPrefEffects::eventFilter(QObject* object, QEvent* event) {
         //   which updates the info box via 'currentRowChanged' signals
         auto pChainList = dynamic_cast<QListView*>(object);
         auto pEffectList = dynamic_cast<QTableView*>(object);
+        // Restore previous selection only if focus was changed with keyboard.
+        // For mouse clicks, that procedure would select the wrong index.
+        QFocusEvent* focEv = static_cast<QFocusEvent*>(event);
```

Comment:
I think we need here a dynamic cast, because we have no guarantee that the QEvent us a QFocusEvent. 
Or alternative check the event type which is faster.

BTW: The casts above can be replaced by a qobject_cast, because they are inherited from QObject.

Reply:
okay, qobject_cast then.
we're already in the `QEvent::FocusIn` branch so it's definitely a QFocusEvent and there's no need to check for null here.

Reply:
wrong, QEvent does not inherit QObject

Reply:
The qobject_cast was the suggestion for the untouched code above. 

In this case I would just add the QEvent::type() check before the static cast to be safe. 
We can't really be sure that it will always be a ficus event. And it is better than a crash when something changes later.

Reply:
ah okay, qobject_cast for the lists. got it.

> In this case I would just add the QEvent::type() check before the static cast to be safe.

? that's already in place
https://github.com/mixxxdj/mixxx/blob/233dcad99db45e0bcba128f7c8940425cbc8359d/src/preferences/dialog/dlgprefeffects.cpp#L308

Reply:
Oh perfect. Sorry for the noise.